pkg/controller/direct/secretmanager/secret_controller.go

```diff
@@ -0,0 +1,381 @@
+	gcpClient *gcp.Client
+	desired   *krm.SecretManagerSecret
+	actual    *secretmanagerpb.Secret
+}
+
+var _ directbase.Adapter = &Adapter{}
+
+func normalizeExternal(ctx context.Context, reader client.Reader, src client.Object, secret *krm.SecretManagerSecret) error {
+	if secret.Spec.Replication != nil {
+		if secret.Spec.Replication.LegacyAutomatic != nil {
+			if secret.Spec.Replication.LegacyAutomatic.CustomerManagedEncryption != nil {
+				kmsKeyRef := secret.Spec.Replication.LegacyAutomatic.CustomerManagedEncryption.KmsKeyRef
+
+				kmsKeyRef, err := refs.ResolveKMSCryptoKeyRef(ctx, reader, src, kmsKeyRef)
+				if err != nil {
+					return err
+				}
+				secret.Spec.Replication.LegacyAutomatic.CustomerManagedEncryption.KmsKeyRef = kmsKeyRef
+			}
+		}
+	}
+	if len(secret.Spec.TopicRefs) != 0 {
+		for _, topicRef := range secret.Spec.TopicRefs {
+			if topicRef.PubSubTopicRef != nil {
+				pubsubRef, err := refs.ResolvePubSubTopic(ctx, reader, src, topicRef.PubSubTopicRef)
+				if err != nil {
+					return err
+				}
+				topicRef.PubSubTopicRef.External = pubsubRef.String()
+			}
+		}
+	}
+	return nil
+
+}
+
+func (a *Adapter) Find(ctx context.Context) (bool, error) {
+	log := klog.FromContext(ctx).WithName(ctrlName)
+	log.V(2).Info("getting SecretManagerSecret", "name", a.id.FullyQualifiedName())
+
+	req := &secretmanagerpb.GetSecretRequest{Name: a.id.FullyQualifiedName()}
+	secretpb, err := a.gcpClient.GetSecret(ctx, req)
+	if err != nil {
+		if direct.IsNotFound(err) {
+			return false, nil
+		}
+		return false, fmt.Errorf("getting SecretManagerSecret %q: %w", a.id.FullyQualifiedName(), err)
+	}
+
+	a.actual = secretpb
+	return true, nil
+}
+
+func MergeMap(a, b map[string]string) map[string]string {
+	for k, v := range b {
+		a[k] = v
+	}
+	return a
+}
+
+func RemoveByPrefixes(a map[string]string, prefixes ...string) map[string]string {
+	for k, _ := range a {
+		for i := range prefixes {
+			if strings.HasPrefix(k, prefixes[i]) {
+				delete(a, k)
+			}
+		}
+	}
+	return a
+}
+
+func UserAnnotations(secret *krm.SecretManagerSecret) map[string]string {
+	annotations := MergeMap(secret.GetAnnotations(), secret.Spec.Annotations)
+	annotations = RemoveByPrefixes(annotations, "cnrm.cloud.google.com", "alpha.cnrm.cloud.google.com")
+	return annotations
+}
+
+func UserLabels(secret *krm.SecretManagerSecret) map[string]string {
+	labels := RemoveByPrefixes(secret.GetLabels(), "cnrm.cloud.google.com")
+	labels["managed-by-cnrm"] = "true"
+	return labels
+}
+
+func (a *Adapter) Create(ctx context.Context, createOp *directbase.CreateOperation) error {
+	u := createOp.GetUnstructured()
+
+	log := klog.FromContext(ctx).WithName(ctrlName)
+	log.V(2).Info("creating Secret", "name", a.id.FullyQualifiedName())
+	mapCtx := &direct.MapContext{}
+
+	desired := a.desired.DeepCopy()
+	resource := SecretManagerSecretSpec_ToProto(mapCtx, &desired.Spec)
+	if mapCtx.Err() != nil {
+		return mapCtx.Err()
+	}
+
+	resource.Annotations = UserAnnotations(desired)
```

Comment:
I assume we support annotations in the TF controller, from `metadata.annotations`, so we aren't free to choose our behaviour here?